api/src/test/java/org/apache/iceberg/metrics/TestDefaultMetricsContext.java

```diff
@@ -83,4 +85,44 @@ public void timer() {
     timer.record(10, TimeUnit.MINUTES);
     Assertions.assertThat(timer.totalDuration()).isEqualTo(Duration.ofMinutes(10L));
   }
+
+  @Test
+  public void intGauge() {
+    MetricsContext metricsContext = new DefaultMetricsContext();
+    Gauge<Integer> gauge = metricsContext.gauge("test", Integer.class);
+    gauge.set(1);
+    Assertions.assertThat(gauge.get()).isEqualTo(1);
+  }
+
+  @Test
+  public void longGauge() {
+    MetricsContext metricsContext = new DefaultMetricsContext();
+    Gauge<Long> gauge = metricsContext.gauge("test", Long.class);
+    gauge.set(1L);
+    Assertions.assertThat(gauge.get()).isEqualTo(1L);
+  }
+
+  @Test
+  public void histogram() {
+    MetricsContext metricsContext = new DefaultMetricsContext();
+    int reservoirSize = 128;
+    Histogram histogram = metricsContext.histogram("test", reservoirSize);
+    for (int i = 0; i < reservoirSize; ++i) {
```

Comment:
nit: I don't think it matters whether we do ++i or i++ here :)

Reply:
unless you intended to add `1..128`, then you'd probably want to rewrite the loop to `for (int i = 1; i <= reservoirSize; i++)`. That would also make it sligthly easier to reason about the different percentiles below. For example, the 50th percentile would then be `Assertions.assertThat(statistics.percentile(0.50)).isEqualTo(64)` instead of `63`

Reply:
sure, I am fine either way. will switch to `1..128`

Reply:
actually switched to `[1, 100]` which will make the assertion number easier to read.